src/mito2/src/read/scan_region.rs

```diff
@@ -697,6 +697,12 @@ impl ScanInput {
             }
         });
     }
+
+    pub(crate) fn total_rows(&self) -> usize {
+        let rows_in_files: usize = self.files.iter().map(|f| f.num_rows()).sum();
+        let rows_in_memtables: usize = self.memtables.iter().map(|m| m.stats().num_rows()).sum();
```

Comment:
We might filter memtables by time range in `ScanRegion::scan_input()`. Will it affect the correctness?

Reply:
> We might filter memtables by time range in `ScanRegion::scan_input()`. Will it affect the correctness?

No, as I commented [here](https://github.com/GreptimeTeam/greptimedb/pull/4545#discussion_r1713643681)